lib/ts/controllers/s-modal.ts

```diff
@@ -153,58 +153,69 @@ namespace Stacks {
         }
 
         /**
-         * Binds tab presses on tabbable items such that tabbing only works within the modal
+         * Gets all elements that could be tabbed to in 
          */
-        private handleFocusableElements() {
-            // get all tabbable items
-            var allTabbables = Array.from(this.modalTarget.querySelectorAll("[href], input, select, textarea, button, [tabindex]"))
+        private getAllTabbables() {
+            return <HTMLElement[]>Array.from(this.modalTarget.querySelectorAll("[href], input, select, textarea, button, [tabindex]"))
                 .filter((el: Element) => el.matches(":not([disabled]):not([tabindex='-1'])"));
+        }
 
-            if (!allTabbables.length) {
-                return;
-            }
-
-            var initialFocus = <HTMLElement>allTabbables[0];
+        private firstVisible(elements: HTMLElement[]) {
+            // https://stackoverflow.com/a/21696585
+            return elements.find(el => el.offsetParent !== null);
+        }
 
-            if (this.hasInitialFocusTarget) {
-                initialFocus = this.initialFocusTarget;
-            }
+        private lastVisible(elements: HTMLElement[]) {
+            return [...elements].reverse().find(el => el.offsetParent !== null);
```

Comment:
nit: Good opportunity to reuse existing code

```suggestion
            return this.firstVisible([...elements].reverse());
```

It sucks that `.reverse` mutates the existing array; it'd be much neater without having to copy the array.